crates/swc_ecma_minifier/src/compress/pure/numbers.rs

```diff
@@ -138,4 +138,33 @@ impl Pure<'_> {
             _ => (),
         }
     }
+
+    pub(super) fn optimize_to_int(&mut self, bin: &mut BinExpr) {
+        let span = bin.span;
+
+        match (bin.op, &mut *bin.left, &mut *bin.right) {
+            (op!("|"), Expr::Bin(e), Expr::Lit(Lit::Num(n)))
+            | (op!("|"), Expr::Lit(Lit::Num(n)), Expr::Bin(e))
+                if matches!(
+                    e.op,
+                    op!("<<") | op!(">>") | op!(">>>") | op!("|") | op!("^") | op!("&")
```

Comment:
`>>> 0` is not NOOP for negative values. `-1 >>> 0 == 0xffffffff`

Reply:
This is for a >>> b | 0

Reply:
Yeah my bad, confused this branch with the one below